hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1602,15 +1602,15 @@ func (r *HostedClusterReconciler) reconcileCAPIManager(ctx context.Context, crea
 
 	// Reconcile CAPI manager deployment
 	capiImage, err := hyperutil.GetPayloadImage(ctx, hcluster, ImageStreamCAPI, pullSecretBytes)
-	if err != nil {
-		return fmt.Errorf("failed to retrieve capi image: %w", err)
-	}
 	if envImage := os.Getenv(images.CAPIEnvVar); len(envImage) > 0 {
 		capiImage = envImage
 	}
 	if _, ok := hcluster.Annotations[hyperv1.ClusterAPIManagerImage]; ok {
 		capiImage = hcluster.Annotations[hyperv1.ClusterAPIManagerImage]
 	}
+	if capiImage == "" && err != nil {
```

Comment:
I don't think we should not be evaluating `err != nil` here. This will still result in an error returned if the image does not exist in the `GetPayloadImage()` called earlier. Instead, maybe we could catch the error above, print the error message as a warning, and finally fail if `capiImage == ""`.

Reply:
It would only return if the `capiImage` is not set from the environment variable or hc annotation. I don't think we would want to continue on with capi manager deployment if `capiImage` is not set, right?

Reply:
Ah sorry! You're absolutely right, I definitely read that wrong. Thanks!